### PR TITLE
AnnotationExtractor: fix tests on windows

### PR DESCRIPTION
**Pull request by staabm**
fixes failing windows test
```
1) Rector\Tests\ChangesReporting\Annotation\AnnotationExtractorTest::testExtractAnnotationFromClass with data set "Class with changelog annotation" ('Rector\Tests\ChangesReporting...ngelog', '@changelog', 'rectorphp/...iew.md')
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'rectorphp/rector@master/docs/rector_rules_overview.md'
+'rectorphp/rector@master/docs/rector_rules_overview.md
'

2) Rector\Tests\ChangesReporting\Annotation\AppliedRectorsChangelogResolver\RectorsChangelogResolverTest::test
Failed asserting that two arrays are identical.
--- Expected
+++ Actual
@@ @@
 Array &0 (
-    'Rector\Tests\ChangesReporting\Annotation\AppliedRectorsChangelogResolver\Source\RectorWithChangelog' => 'rectorphp/rector@master/docs/rector_rules_overview.md'
+    'Rector\Tests\ChangesReporting\Annotation\AppliedRectorsChangelogResolver\Source\RectorWithChangelog' => 'https://github.com/rectorphp/rector/blob/master/docs/rector_rules_overview.md\r
+'
 )
 ```

### CODE REVIEW

**Comment by staabm** [comment]
ready to land

File: packages/ChangesReporting/Annotation/AnnotationExtractor.php [hunk 1]
```diff
@@ -28,6 +28,10 @@ public function extractAnnotationFromClass(string $className, string $annotation
         // @see https://regex101.com/r/oYGaWU/1
         $pattern = '#' . preg_quote($annotation, '#') . '\s+(?<content>.*?)$#m';
         $matches = Strings::match($docComment, $pattern);
-        return $matches['content'] ?? null;
+
+        if ($matches && $matches['content']) {
```
Comment (staabm) [inline packages/ChangesReporting/Annotation/AnnotationExtractor.php:32]: hmm?

```suggestion
        if ($matches) {
```

**Comment by TomasVotruba** [comment]
What exactly is going on here? 

This looks like a workaround for a bug in regex. We should fix the bug in regex instead. Ideally with link to verification on https://regex101.com/

**Comment by staabm** [comment]
It seems the regex is matching \r\n and therefore the \r\n is part of the matched url at the very end

Actually it could be reduced to a `rtrim`. I see no easy way to support a multi line regex fixing the issue at hand.

**Comment by TomasVotruba** [comment]
If `rtrim()` does the same work, let's go for it :+1:

**Comment by staabm** [comment]
Fix verified. Ready to land

File: packages/ChangesReporting/Annotation/AnnotationExtractor.php [hunk 2]
```diff
@@ -29,8 +29,8 @@ public function extractAnnotationFromClass(string $className, string $annotation
         $pattern = '#' . preg_quote($annotation, '#') . '\s+(?<content>.*?)$#m';
         $matches = Strings::match($docComment, $pattern);
 
-        if ($matches && $matches['content']) {
-            return trim($matches['content']);
+        if ($matches && isset($matches['content'])) {
```
Comment (TomasVotruba) [inline packages/ChangesReporting/Annotation/AnnotationExtractor.php]: What is in `$matches` exactly? The `Strings::match()` returns array only if named arguments are found. So checking it here again does not make sense.
There must be some bug in regular expression.

File: packages/ChangesReporting/Annotation/AnnotationExtractor.php [hunk 3]
```diff
@@ -28,6 +28,10 @@ public function extractAnnotationFromClass(string $className, string $annotation
         // @see https://regex101.com/r/oYGaWU/1
         $pattern = '#' . preg_quote($annotation, '#') . '\s+(?<content>.*?)$#m';
         $matches = Strings::match($docComment, $pattern);
-        return $matches['content'] ?? null;
+
+        if ($matches && isset($matches['content'])) {
```
Comment (staabm) [inline packages/ChangesReporting/Annotation/AnnotationExtractor.php]: yeah.. the nette method has not good phpdocs, and therefore its not clear when it returns what.

I gues the `isset` is not required.

```suggestion
        if ($matches) {
```

**Comment by staabm** [comment]
regarding the regex.

question is, why does https://3v4l.org/nU270 match the string without a ending newline, but https://3v4l.org/G9NNG contains the ending newline in the matched string

**Comment by staabm** [comment]
fixed the regex and reverted the `rtrim` workaround. ready to land.

**Comment by staabm** [comment]
@TomasVotruba good to go

**Comment by TomasVotruba** [comment]
Thank you :+1: Elegant fix

**Pull request merged**